### PR TITLE
Big VMs

### PR DESCRIPTION
**Pull request by joker-official**
These are a couple of changes for big VMs as talked about with @Scsabiii and @galkindmitrii .

Still missing test coverage and probably fixes for pep8/existing tests.

### CODE REVIEW

File: nova/scheduler/filters/bigvm_filter.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        except ValueError:
+            LOG.exception("%(host_state)s has an invalid value for "
+                          "%(aggregate_key)s: %(aggregate_val)s. Only "
+                          "integers are supported.",
+                          {'host_state': host_state,
+                           'aggregate_key': _AGGREGATE_KEY,
+                           'aggregate_val': aggregate_val})
+            return False
+
+        max_ram_percent = self._get_max_ram_percent(requested_ram_mb,
```
Comment (galkindmitrii) [inline nova/scheduler/filters/bigvm_filter.py]: I think you wanted to call '_compute_max_ram_percent' here?

**Comment by galkindmitrii** [comment]
lgtm, but I'd like to see unit-tests for new filter here :)

File: nova/scheduler/filters/bigvm_filter.py (hunk 1 above)
Comment (joker-official) [reply nova/scheduler/filters/bigvm_filter.py]: Yeah, I renamed it and forgot to rename the function. Unit-tests showed me my error :D

File: nova/tests/unit/scheduler/filters/test_bigvm_filters.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                 'total_usable_ram_mb': self.hv_size})
+        self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_big_vm_without_enough_ram(self, agg_mock):
+        # there's enough RAM available in the cluster but not enough (~50 % of
+        # the requested size on average
+        # 12 hosts (bigvm + 1 GB size)
+        # 11 big VM + some smaller (12 * 1 GB) already deployed
+        # -> still bigvm_mb left, but ram utilization ratio of all hosts is too
+        # low
```
Comment (galkindmitrii) [inline nova/tests/unit/scheduler/filters/test_bigvm_filters.py]: is too high?

Comment (joker-official) [reply nova/tests/unit/scheduler/filters/test_bigvm_filters.py]: Of course m)

File: nova/tests/unit/scheduler/filters/test_bigvm_filters.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_big_vm_with_non_int_aggregate(self, agg_mock):
+        agg_mock.return_value = {'hv_size_mb': ['foo']}
+        spec_obj = objects.RequestSpec(
+            flavor=objects.Flavor(memory_mb=CONF.bigvm_mb))
+        host = fakes.FakeHostState('host1', 'compute', {})
+        self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_big_vm_without_engough_ram(self, agg_mock):
```
Comment (galkindmitrii) [inline nova/tests/unit/scheduler/filters/test_bigvm_filters.py]: typo: enough

File: nova/tests/unit/scheduler/filters/test_bigvm_filters.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        agg_mock.return_value = {'hv_size_mb': [self.hv_size]}
+        spec_obj = objects.RequestSpec(
+            flavor=objects.Flavor(memory_mb=self.hv_size + 1))
+        host = fakes.FakeHostState('host1', 'compute', {})
+        self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_big_vm_with_engough_ram(self, agg_mock):
```
Comment (galkindmitrii) [inline nova/tests/unit/scheduler/filters/test_bigvm_filters.py]: typo: enough

File: nova/tests/unit/scheduler/filters/test_bigvm_filters.py (hunk 3 above)
Comment (joker-official) [reply nova/tests/unit/scheduler/filters/test_bigvm_filters.py]: fixed. thanks.

File: nova/scheduler/filters/bigvm_filter.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+class BigVmBaseFilter(filters.BaseHostFilter):
+
+    _HV_SIZE_CACHE = {}
+    _HV_SIZE_CACHE_RETENTION_TIME = 10 * 60
+
+    RUN_ON_REBUILD = False
+
+    def _get_hv_size(self, host_state):
+        # expire the cache every 10min
+        LOG.info(self._HV_SIZE_CACHE)
```
[the remark this answers is not in the document]
Comment (joker-official) [reply nova/scheduler/filters/bigvm_filter.py]: Haha, thanks. That was debug output for testing.

**Pull request merged**